Add role templates path to jinja2 loader paths.

wkf opened the pull request:
Lately, I've found myself wanting to include 'base' templates in a role, and then extend them with more specfic templates outside the roles. This makes the roles more reusable, and can be accomplished now by using a path like `roles/nginx/templates/vhost.j2`. Where it starts to break down is when you have altered `roles_path`, since now the nginx role might not be in the `roles` directory. So, in order to avoid coupling the path to the layout of your directory, I've made a small change to search the role's templates directory when using `extends`.

Commit:
Add role templates path to jinja2 loader.

mpdehaan commented:
Thanks for this, I don't see a problem with the roles/template path being able to be used easily from template includes.

There are some things here I'd like to see corrected.

A variable like "templates", if it exists, seems to imply an array.

Rather than "template_from_file" taking a role_paths parameter, perhaps it should take a "search_paths" parameter.

Variables should default to None if not set, not empty string.

I'm also a little bit unclear on this part and what this does:

```
role_templates != load_paths[1]:
```

Also, in the future, please do not submit commits that only affect whitespace.   This breaks line attribution and really doesn't have any effects we are seeking in commit history.

Thanks!

wkf commented:
Thanks for the feedback @mpdehaan... I've dropped the whitespace commit, and made the changes you requested. `role_templates != load_paths[1]:` was an attempt to only add the item to the list if it wasn't there already. I've updated to code to make the intent more clear. Thanks!

amenonsen commented:
@wkf any chance you could rebase this to apply to current devel?

sivel commented:
@wkf,

Hi!

Thank you very much for your submission to Ansible.  It sincerely means a lot to us.

Unfortunately it looks like the code underneath has changed since this was submitted.  Specifically this change was written against a code path that no longer exists in Ansible.

We would like to ask if you could either update the code in this pull request to apply against the current state of Ansible or close this pull request. If we do not hear back from you, we will close this pull request out shortly.

Thank you once again for this and your interest in Ansible!

amenonsen commented:
Closing per the comment above.
